File: sol/2696/2696.py
class Solution:
    def minLength(self, s: str) -> int:
        st = []
        for c in s:
            if len(st) >= 1 and ((st[-1] == 'A' and c == 'B') or (st[-1] == 'C' and c == 'D')):
                st.pop()
            else:
                st.append(c)
        return len(st)

File: sol/2696/2696.cpp
struct Solution {
    int minLength(string &s) {
        vector<char> st;
        for (char c : s){
            if (!st.empty() && ((st.back() == 'A' && c == 'B') || (st.back() == 'C' && c == 'D'))){
                st.pop_back();
            } else {
                st.push_back(c);
            }
        }
        return st.size();
    }
};
